admin/includes/classes/observers/class.admin_submenus.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+if (!defined('IS_ADMIN_FLAG') || IS_ADMIN_FLAG !== true) {
+    die('Illegal Access');
+}
+
+class zcObserverAdminSubmenus extends base
+{
+    public function __construct()
+    {
+        $this->attach($this, ['NOTIFY_LANGUAGE_CHANGE_REQUESTED_BY_ADMIN_VISITOR']);
+    }
+
+    public function update(&$class, $eventID, $lang, &$olan)
+    {
+        $admin_submenus = [];
+        $table_type = [];
+
+        $conn = new mysqli(constant('DB_SERVER'), DB_SERVER_USERNAME, DB_SERVER_PASSWORD, constant('DB_DATABASE'));
+        
+        $lang_dir_name = DIR_WS_LANGUAGES . $olan->language['directory'] . '/admin_submenus';
+        if (is_dir($lang_dir_name)) {
+            $dir_content = array_diff(scandir($lang_dir_name), array('..', '.'));
+            foreach($dir_content as $key => $filename) {
+                if (is_file($lang_dir_name . '/' . $filename) && strpos($filename, 'admin_menus') === 0) { // checking for files starting by 'admin_menus' to include them as they should contain '$admin_submenus' table data.
+                    include($lang_dir_name . '/' . $filename);
+                }
+            }
+        }
+        
+        $plugin_infos = $conn->query('SELECT unique_key, version FROM plugin_control WHERE 1'); // check fo installed encapsulated plugins
```

Comment:
This could be a regular `$result = $db->Execute()` call, which then iterates `foreach ($result as $plugin)`

Reply:
Done!